src/gradient_aware_harmonisation/add_cubic.py

```diff
@@ -0,0 +1,199 @@
+    coeffs_r = coeffs[::-1]
+
+    # There are less compute and memory intense ways to do this,
+    # but we're not that worried about that right now.
+    store = np.zeros((n + 1, n + 1))
+    for i in range(n):
+        store[i, 0] = coeffs_r[n - i - 1] * shift ** (n - i - 1)
+        store[i, i + 1] = coeffs_r[n] * shift**n
+
+    for j in range(n):
+        for i in range(j + 1, n + 1):
+            store[i, j + 1] = store[i - 1, j] + store[i - 1, j + 1]
+
+    res_r = np.zeros(coeffs.size)
+    for i in range(n):
+        res_r[i] = store[n, i + 1] / shift**i
+
+    res_r[n] = coeffs_r[n]
+
+    # Return back to expected convention
+    res = res_r[::-1]
+    return res
+
+
+def harmonise_splines_add_cubic(
+    diverge_from: Spline,
+    harmonisee: Spline,
+    harmonisation_time: float | int,
+    convergence_time: float | int,
+) -> Spline:
+    """
+    Generate cubic spline
+
+    The cubic spline interpolates between harmonised
+    spline and harmonisee
+
+    Parameters
+    ----------
+    diverge_from
+        Spline whose value and first derivative at the
+        harmonization time match the target spline values.
+
+    harmonisee
+        Spline that the cubic spline should converge to
+        and match at and after the convergence time.
+
+    harmonisation_time
+        Time point at which cubic spline should be
+        matched to the target.
+
+    convergence_time
+        Time point at which the cubic spline should
+        converge to harmonisee.
+
+    Returns
+    -------
+    :
+        cubic spline
+
+    Examples
+    --------
+    >>> from gradient_aware_harmonisation.timeseries import Timeseries
+    >>> from gradient_aware_harmonisation.add_cubic import harmonise_splines_add_cubic
+    >>> import numpy as np
+    >>> import pandas as pd
+    >>> import matplotlib.pyplot as plt
+
+    >>> harmonisation_time = 3.2
+    >>> convergence_time = 3.8
+    >>> time = np.arange(3.0, 4.1, step=0.1)
+
+    >>> harmonised = Timeseries(time_axis=time, values=-0.5 * time + 1)
+    >>> harmonisee = Timeseries(time_axis=time, values=0.2 * time**3 - 5)
+    >>> cubic_spline = harmonise_splines_add_cubic(
+    ...     diverge_from=harmonised.to_spline(),
+    ...     harmonisee=harmonisee.to_spline(),
+    ...     harmonisation_time=harmonisation_time,
+    ...     convergence_time=convergence_time,
+    ... )
+
+    >>> _, ax = plt.subplots(figsize=(6, 3))
+    >>> for y, name in zip(
+    ...     [harmonised.values, harmonisee.values, cubic_spline(time)],
+    ...     ["diverge_from", "harmonisee", "cubic_spline"],
+    ... ):
+    ...     ax.plot(time, y, label=name)  # doctest: +SKIP
+    >>> ax.axvline(harmonisation_time, color="black", linestyle="--")  # doctest: +SKIP
+    >>> ax.axvline(convergence_time, color="black", linestyle="--")  # doctest: +SKIP
+    >>> ax.legend()  # doctest: +SKIP
+    >>> plt.show()  # doctest: +SKIP
+
+    ![](../../../assets/images/cubic_spline.png)
```

Comment:
I'm super impressed you managed to make this work. I'm happy to try this out, my current concern is only how to keep it up to date. By the look of it, `../../../assets/images/cubic_spline.png` is a static file and would only update if we remember, rather than automatically if we changed the behaviour of the function. (This concern is why I'd probably just leave plotting examples to the notebooks, but as I said happy to try this way out and see how it goes, it's not going to cause any big headaches I don't think.)